### PR TITLE
Adds CBMC proofs for aws_cryptosdk_priv_hdr_parse_{header_version, message_type}

### PR DESCRIPTION
**Pull request by tegansb**
*Description of changes:*
Adds harness, Makefile and yaml for aws_cryptosdk_priv_hdr_parse_header_version and aws_cryptosdk_priv_hdr_parse_message_type. 

Proofs are for split version of hdr_parse in https://github.com/aws/aws-encryption-sdk-c/pull/641. 
Uses ensures, is bounded and validity functions from https://github.com/aws/aws-encryption-sdk-c/pull/632. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_message_type/aws_cryptosdk_priv_hdr_parse_message_type_harness.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    __CPROVER_assume(hdr->enc_ctx.p_impl != NULL);
+    ensure_hash_table_has_valid_destroy_functions(&hdr->enc_ctx);
+    size_t empty_slot_idx;
+    __CPROVER_assume(aws_hash_table_has_an_empty_slot(&hdr->enc_ctx, &empty_slot_idx));
+
+    __CPROVER_assume(pcursor != NULL);
+    __CPROVER_assume(aws_byte_cursor_is_bounded(pcursor, MAX_BUFFER_SIZE));
+    ensure_byte_cursor_has_allocated_buffer_member(pcursor);
+    __CPROVER_assume(aws_byte_cursor_is_valid(pcursor));
```
Comment (alex-chew) [inline verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_message_type/aws_cryptosdk_priv_hdr_parse_message_type_harness.c:43]: Same as https://github.com/aws/aws-encryption-sdk-c/pull/644#discussion_r538953400 - can we pull these out into a helper function?

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    ensure_hash_table_has_valid_destroy_functions(&hdr->enc_ctx);
+    size_t empty_slot_idx;
+    __CPROVER_assume(aws_hash_table_has_an_empty_slot(&hdr->enc_ctx, &empty_slot_idx));
+
+    __CPROVER_assume(pcursor != NULL);
+    __CPROVER_assume(aws_byte_cursor_is_bounded(pcursor, MAX_BUFFER_SIZE));
+    ensure_byte_cursor_has_allocated_buffer_member(pcursor);
+    __CPROVER_assume(aws_byte_cursor_is_valid(pcursor));
+
+    __CPROVER_assume(header_version != NULL);
```
Comment (alex-chew) [inline verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c:46]: Same as https://github.com/aws/aws-encryption-sdk-c/pull/644#discussion_r538953400 - can we pull these out into a helper function?

**Review by alex-chew — APPROVED** [review]
Overall LGTM. I think we can pull some assumptions/preconditions out into a helper function, but it's not a blocker for this PR.

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_message_type/aws_cryptosdk_priv_hdr_parse_message_type_harness.c (hunk 1 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_message_type/aws_cryptosdk_priv_hdr_parse_message_type_harness.c]: Addressed with new hdr_setup function!

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c (hunk 2 above)
Comment (tegansb) [reply verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c]: Addressed with new hdr_setup function!

File: verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#include <aws/cryptosdk/header.h>
+#include <aws/cryptosdk/private/header.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+void aws_cryptosdk_priv_hdr_parse_header_version_harness() {
+    /* Nondet Input */
+    struct aws_cryptosdk_hdr *hdr   = hdr_setup(MAX_TABLE_SIZE, MAX_EDK_LIST_ITEMS, MAX_BUFFER_SIZE);
```
Comment (feliperodri) [inline verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_header_version/aws_cryptosdk_priv_hdr_parse_header_version_harness.c:25]: Sure it's way cleaner.

**Review by feliperodri — APPROVED** [review]
LGTM.

**Pull request merged**